### PR TITLE
US-547544 : Hazelcast enablement prconfig added

### PR DESCRIPTION
**Pull request by vnihal72**
US-547544 : Hazelcast enablement prconfig added

### CODE REVIEW

**Review by taz-pega-work — COMMENTED** [review]
please comment or confirm why v4 is listed. For the latest deployments, I though clients were supposed to run v5 and later??

**Review by MadhuriArugula — APPROVED** [review]

**Comment by vnihal72** [comment]
> please comment or confirm why v4 is listed. For the latest deployments, I though clients were supposed to run v5 and later??

It is default value which will be overridden by Pega Helm chart accordingly based on Pega version.

File: tomcat-conf/Catalina/localhost/prweb.xml [hunk 1]
```diff
@@ -73,6 +73,10 @@
   <Environment name="prconfig/dsm/services/stream/pyBaseLogPath" value="/opt/pega/kafkadata/data" type="java.lang.String" />
   <Environment name="prconfig/cloud/isNodeAgnosticAdminStudio" value="true" type="java.lang.String"/>
 
+  {{ if .Env.HZ_VERSION }}
+  <Environment name="prconfig/cluster/hazelcast/{{ .Env.HZ_VERSION }}/enabled" value="true" type="java.lang.String"/>
+  {{ end }}
```
Comment (kishorv10) [inline tomcat-conf/Catalina/localhost/prweb.xml:78]: Is `prconfig/cluster/clientserver/server/hostname` not required to set? AFAIK this value is different for v4 to v5?

File: Dockerfile [hunk 2]
```diff
@@ -150,6 +150,7 @@ ENV PEGA_SEARCH_URL=
 
 # Configure hazelcast. By default, hazelcast runs in embedded mode.
 ENV HZ_CLIENT_MODE=false \
+    HZ_VERSION=v4 \
```
Comment (kishorv10) [inline Dockerfile:153]: Why the default value set to `v4`?

File: tomcat-conf/Catalina/localhost/prweb.xml (hunk 1 above)
Comment (Saurabh-16) [reply tomcat-conf/Catalina/localhost/prweb.xml:78]: Don't we need the below condition at this point? This flag should only be passed when we are running Hz in c/s mode.
`{{ if isTrue .Env.HZ_CLIENT_MODE }}`

Comment (vnihal72) [reply tomcat-conf/Catalina/localhost/prweb.xml:78]: As discussed over the call All C/s related configs are already in prweb.

File: Dockerfile (hunk 2 above)
Comment (vnihal72) [reply Dockerfile:153]: As discussed over the call , v4 is default as we had in pega helm chart before.
This value will be overridden by Pega Helm chart accordingly based on Pega version.

File: tomcat-conf/Catalina/localhost/prweb.xml (hunk 1 above)
Comment (kishorv10) [reply tomcat-conf/Catalina/localhost/prweb.xml:78]: After discussing with @vnihal72 , I have learned that the changes to this parameter will be made in the pull request for the Helm charts

**Comment by vnihal72** [comment]
> Now that, prweb.xml contain v5 hz related params all the time, Is there any cleanup required in pega-helm-charts and hz backing service chart? I remember few fields like skipRestart, etc.. are added in pega-helm-charts to avoid restart, Can those be simplified?

As discussed over the call , These changes are part of Pega Helm chart , will merge after the testing through CloudK GOC

File: Dockerfile (hunk 2 above)
Comment (kishorv10) [reply Dockerfile:153]: After discussing with @vnihal72 , I have learned that earlier this default value is set in helm charts, same is moved to docker,

**Review by kishorv10 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by taz-pega-work — APPROVED** [review]
in future fix, please update this comment in the Dockerfile since it will be confusing for any clients looking in the contents>>
# Configure hazelcast. By default, hazelcast runs in embedded mode.

**Pull request merged**